aws/nullClean.jq

```diff
@@ -0,0 +1,11 @@
+# Apply f to composite entities recursively, and to atoms
+def walk(f):
+  . as $in
+  | if type == "object" then
+      reduce keys[] as $key
+        ( {}; . + { ($key):  ($in[$key] | walk(f)) } ) | f
+  elif type == "array" then map( walk(f) ) | f
+  else f
+  end;
+
+walk( if type == "object" then with_entries(select(.value != null)) else . end )
```

Comment:
In testing I found this was stripping all the nulls except where the objects value was a null array - ```"test" : [null]```. Unfortunately this is how arrays that aren't specified are output from an ARM deployment.

```
{
  "locations": [
    null
  ],
  "resourceType": "storageAccounts/blobServices"
},
```

Amending to the following worked:
```
walk( if type == "object" then with_entries(select(.value != null and .value != [null])) else . end )
 ```

Reply:
Thanks moved this PR into #1124